tests/unit/db/database_tests.py

```diff
@@ -0,0 +1,480 @@
+        self.assertEqual(doc.get('name'), 'julia')
+        self.assertEqual(doc.get('age'), 1)
+
+        # Test __get_item__ for design document
+        ddoc = self.db['_design/ddoc01']
+        self.assertIsInstance(ddoc, DesignDocument)
+        self.assertTrue(ddoc, expected_ddoc)
+
+    def test_document_iteration_under_fetch_limit(self):
+        """
+        Test __iter__ works as expected when the number of documents in
+        the database is less than the database fetch limit
+        """
+        docs = []
+
+        # Check iterating when no documents exist
+        for doc in self.db:
+            self.fail('There should be no documents in the database yet!!')
+        # Check that iteration yields appropriate contents
+        self.populate_db_with_documents(3)
+        age = 0
+        for doc in self.db:
+            self.assertIsInstance(doc, Document)
+            self.assertEqual(doc['_id'], 'julia{0:03d}'.format(age))
+            self.assertTrue(doc['_rev'].startswith('1-'))
+            self.assertEqual(doc['name'], 'julia')
+            self.assertEqual(doc['age'], age)
+            docs.append(doc)
+            age += 1
+        self.assertEqual(len(docs), 3)
+        # Check that the local database object has been populated
+        # with the appropriate documents
+        expected_keys = []
+        for i in range(3):
+            expected_keys.append('julia{0:03d}'.format(i))
+        self.assertTrue(all(x in self.db.keys()for x in expected_keys))
+        for id in self.db.keys():
+            doc = self.db.get(id)
+            self.assertIsInstance(doc, Document)
+            self.assertEqual(doc['_id'], id)
+            self.assertTrue(doc['_rev'].startswith('1-'))
+            self.assertEqual(doc['name'], 'julia')
+            self.assertEqual(doc['age'], int(id[len(id) - 3 : len(id)]))
+
+    def test_document_iteration_over_fetch_limit(self):
+        """
+        Test __iter__ works as expected when the number of documents in
+        the database is more than the database fetch limit
+        """
+        docs = []
+        # Check iterating when no documents exist
+        for doc in self.db:
+            self.fail('There should be no documents in the database yet!!')
+        # Check that iteration yields appropriate contents
+        self.populate_db_with_documents(103)
+        age = 0
+        for doc in self.db:
+            self.assertIsInstance(doc, Document)
+            self.assertEqual(doc['_id'], 'julia{0:03d}'.format(age))
+            self.assertTrue(doc['_rev'].startswith('1-'))
+            self.assertEqual(doc['name'], 'julia')
+            self.assertEqual(doc['age'], age)
+            docs.append(doc)
+            age += 1
+        self.assertEqual(len(docs), 103)
+        # Check that the local database object has been populated
+        # with the appropriate documents
+        expected_keys = []
+        for i in range(103):
+            expected_keys.append('julia{0:03d}'.format(i))
```

Comment:
expected_keys = ['julia{0:03d}'.format(i) for i in xrange(103)] is more pythonic, though not really a big deal here, so just a suggestion